pyramid/predicates.py

```diff
@@ -132,14 +132,19 @@ def __call__(self, context, request):
 class AcceptPredicate(object):
     def __init__(self, val, config):
         self.val = val
+        if '*' in self.val:
+            raise ConfigurationError(
+                '"accept" predicate does not accept media ranges, only '
+                'specific media types',
+            )
 
     def text(self):
         return 'accept = %s' % (self.val,)
 
     phash = text
 
     def __call__(self, context, request):
-        return self.val in request.accept
```

Comment:
As @whiteroses mentioned, I don't think this allowed you to pass in `text/*` for example...

Reply:
You can pass in `text/*` I think, but you'd get a `ValueError` from the predicate calling `__contains__` if the header was not present in the request or was the empty string &mdash; you can see it in [this test](https://github.com/whiteroses/pyramid/blob/before-webob-changes/pyramid/tests/test_integration.py#L895) from the [before-webob-changes](https://github.com/whiteroses/pyramid/tree/before-webob-changes) branch from [here](https://github.com/Pylons/pyramid/pull/3326#issuecomment-414161922).

Reply:
this works properly now, contingent on webob master + https://github.com/Pylons/webob/pull/376